server/src/main/java/org/opensearch/indices/recovery/MultiFileWriter.java

```diff
@@ -156,7 +156,10 @@ private void innerWriteFileChunk(StoreFileMetadata fileMetadata, long position,
                 + temporaryFileName
                 + "] in "
                 + Arrays.toString(store.directory().listAll());
-            store.directory().sync(Collections.singleton(temporaryFileName));
+            // With Segment Replication, we will fsync after a full commit has been received.
```

Comment:
The `if` condition gating this code block has `||` on lastChunk. This means this will execute on last chunk of every file ?

```
if (indexOutput.getFilePointer() >= fileMetadata.length() || lastChunk)
```

Reply:
Ignore comment above, code below executes for document replication type. Segrep sync is handled outside of this block.